integration/hive/src/main/java/org/apache/carbondata/hive/MapredCarbonOutputFormat.java

```diff
@@ -92,6 +95,14 @@ public void checkOutputSpecs(FileSystem fileSystem, JobConf jobConf) throws IOEx
     }
     String tablePath = FileFactory.getCarbonFile(carbonLoadModel.getTablePath()).getAbsolutePath();
     TaskAttemptID taskAttemptID = TaskAttemptID.forName(jc.get("mapred.task.id"));
+    // taskAttemptID will be null when the insert job is fired from presto. Presto send the JobConf
+    // and since presto does not use the MR framework for execution, the mapred.task.id will be
+    // null, so prepare a new ID.
+    if (taskAttemptID == null) {
+      SimpleDateFormat formatter = new SimpleDateFormat("yyyyMMddHHmm");
+      String jobTrackerId = formatter.format(new Date());
+      taskAttemptID = new TaskAttemptID(jobTrackerId, 0, TaskType.MAP, 0, 0);
```

Comment:
Here `taskAttemptID ` is `TaskAttemptID` object. Since for every writer it creates new task, there should be no problem. We get the jobconf from presto, we prepare the taskattemptid just for writer close purpose and initialize, so it should be fine i guess. what you think?

With respect to ORC writer if you see, ORC uses the different `FIleOutPutFormat `from `mapred `package, we use `mapreduce `package, In mapred, taskcontext is not used, so they are not using this.

Reply:
ok, If this task number is used in file name, in case of non-transactional  concurrent write. two files can have same file name leading to many issues. so, I suggested UUID. you can check again.

Reply:
Also please check filenames while testing whether segment id and other info is proper in the file name created by presto.

Reply:
> Also please check filenames while testing whether segment id and other info is proper in the file name created by presto.

`Fact/Part0/Segment_10/part-0-0_batchno0-0-10-1603260474337.snappy.carbondata`,  `Fact/Part0/Segment_10/10_1603260475282.carbonindexmerge`

These are indexmerge and carbon file inside segment for segment 10, so naming is fine.

Reply:
> ok, If this task number is used in file name, in case of non-transactional concurrent write. two files can have same file name leading to many issues. so, I suggested UUID. you can check again.

I set the taskID to loadmodel only of the mapred.task.id is present and taskAttempt is not null, if null i dont set taskID to loadmodel, when we call super.getRecordWriter, CarbonTableOutputFormat will set load model based on DEFAULT_TASK_NO. Please have a look